packages/runner/lib/sdk/persist.ts

```diff
@@ -0,0 +1,312 @@
+import https from 'node:https';
+import type { AxiosInstance, AxiosRequestConfig, AxiosResponse } from 'axios';
+import axios, { AxiosError, isAxiosError } from 'axios';
+import { getPersistAPIUrl } from '@nangohq/shared';
+import { getUserAgent } from '@nangohq/node';
+import { httpRetryStrategy, retryWithBackoff, Ok, Err } from '@nangohq/utils';
+import { logger } from '../logger.js';
+import type { Result } from '@nangohq/utils';
+import type { CursorOffset, DeleteRecordsSuccess, GetCursorSuccess, MergingStrategy, PostRecordsSuccess, PutRecordsSuccess } from '@nangohq/types';
+
+export interface PersistApi {
+    saveLog({ environmentId, data }: { environmentId: number; data: string }): Promise<Result<void>>;
+    saveRecords<T = any>({
+        model,
+        records,
+        environmentId,
+        providerConfigKey,
+        connectionId,
+        nangoConnectionId,
+        syncId,
+        syncJobId,
+        activityLogId,
+        merging
+    }: {
+        model: string;
+        records: T[];
+        environmentId: number;
+        providerConfigKey: string;
+        connectionId: string;
+        nangoConnectionId: number;
+        syncId: string;
+        syncJobId: number;
+        activityLogId: string;
+        merging: MergingStrategy;
+    }): Promise<Result<PostRecordsSuccess>>;
+    updateRecords<T = any>({
+        model,
+        records,
+        environmentId,
+        providerConfigKey,
+        nangoConnectionId,
+        connectionId,
+        syncId,
+        syncJobId,
+        activityLogId,
+        merging
+    }: {
+        model: string;
+        records: T[];
+        environmentId: number;
+        providerConfigKey: string;
+        connectionId: string;
+        nangoConnectionId: number;
+        syncId: string;
+        syncJobId: number;
+        activityLogId: string;
+        merging: MergingStrategy;
+    }): Promise<Result<PutRecordsSuccess>>;
+    deleteRecords<T = any>({
+        model,
+        records,
+        environmentId,
+        providerConfigKey,
+        connectionId,
+        nangoConnectionId,
+        syncId,
+        syncJobId,
+        activityLogId,
+        merging
+    }: {
+        model: string;
+        records: T[];
+        environmentId: number;
+        providerConfigKey: string;
+        connectionId: string;
+        nangoConnectionId: number;
+        syncId: string;
+        syncJobId: number;
+        activityLogId: string;
+        merging: MergingStrategy;
+    }): Promise<Result<DeleteRecordsSuccess>>;
+    getCursor({
+        environmentId,
+        nangoConnectionId,
+        model,
+        offset
+    }: {
+        environmentId: number;
+        nangoConnectionId: number;
+        model: string;
+        offset: CursorOffset;
+    }): Promise<Result<GetCursorSuccess>>;
+}
```

Comment:
cusiosity: Why make an interface when you could just have the class used as a type?

Reply:
I believe you can just use the class as a regular interface